File: python/src/shapes/base/models/square.py
from typing import final

from shapes.base.commons import Dimensional


@final
class Square(Dimensional):
    def __init__(self, side: float):
        dim_one: float = side if side > 0 else 1
        self.side = dim_one

    def area(self) -> float:
        return pow(self.side, 2)

    def perimeter(self) -> float:
        return 4 * self.side

File: ruby/lib/shapes/base/models/square.rb
# frozen_string_literal: true

require "shapes/base/commons"

class Square
    include Dimensional

    attr_accessor :side

    def initialize(side)
        dim_one = side >= 0 ? side : 0
        @side = dim_one
    end

    def area
        @side**2
    end

    def perimeter
        @side * 4
    end
end
